### ISSUE TITLE
Twilight forest keeps spawning the same boss

### ISSUE THREAD

**Issue by bluelightning32** [SpongePowered/SpongeForge#2682]
**I am currently running**
- SpongeForge version: spongeforge-1.12.2-2768-7.1.6-RC3643.jar
- Forge version: forge-1.12.2-14.23.5.2822-universal.jar
- Java version: java-1.8.0-openjdk-1.8.0.201.b09-2.fc29.x86_64
- Operating System: Fedora 29

- Modpack: DD&SS 4.9
- Plugins/Mods:
```
[18:17:25] [Server thread/INFO] [minecraft/DedicatedServer]: Plugins (9): Minecraft, Minecraft Coder Pack, SpongeAPI, Sponge, Inventory Sync, JdaLibraryLoader, LuckPerms, UltimateChat, spark
```

**Issue Description**
When approaching a boss location in the twilight forest, it keeps spawning the same boss over and over until the FPS drops so low you can't move. My guess is that the TF spawner isn't breaking like it is supposed to after it spawns the first boss.

![image](https://user-images.githubusercontent.com/21202621/56072798-a4313a80-5d4f-11e9-8be3-6c8923cd2012.png)

The debug log has this message a lot, but I think it is happening after the nagas were spawned, not before:
```
[18:12:15] [Server thread/WARN] [Sponge]: EntityTFNagaSegment['unknown'/33996, l='DIM7', x=-272.13, y=32.96, z=-755.55] is not in this chunk, skipping save. This is a bug fix to a vanilla bug. Do not report this to Sponge or Forge please.
[18:12:15] [Server thread/WARN] [Sponge]: EntityTFNagaSegment['unknown'/34338, l='DIM7', x=-255.30, y=32.06, z=-752.96] is not in this chunk, skipping save. This is a bug fix to a vanilla bug. Do not report this to Sponge or Forge please.
[18:12:15] [Server thread/WARN] [Sponge]: EntityTFNagaSegment['unknown'/34419, l='DIM7', x=-255.96, y=32.54, z=-765.83] is not in this chunk, skipping save. This is a bug fix to a vanilla bug. Do not report this to Sponge or Forge please.
[18:12:15] [Server thread/WARN] [Sponge]: EntityTFNagaSegment['unknown'/34013, l='DIM7', x=-287.73, y=32.00, z=-752.72] is not in this chunk, skipping save. This is a bug fix to a vanilla bug. Do not report this to Sponge or Forge please.
```

I pregenerated this work using `'Chunk Pregenerator V1.12-2.1.jar'`. I've tried pregenerating it directly in sponge forge, and I've tried pregenerating it with forge then copying the DIM7 folder to sponge forge.

**Comment by bluelightning32** [comment]
I reset the dimension and reproduced it without pregenning.

**Comment by bluelightning32** [comment]
Reproduces in spongeforge-1.12.2-2768-7.1.6-RC3634.jar.

**Comment by bluelightning32** [comment]
Reproduces in spongeforge-1.12.2-2768-7.1.6-RC3634.jar without the phosphor mod.

**Comment by bluelightning32** [comment]
No repro in spongeforge-1.12.2-2768-7.1.6-RC3596.jar without the phosphor mod (phosphor is incompatible with that old of a spongeforge version).

**Comment by bluelightning32** [comment]
No repro in spongeforge-1.12.2-2768-7.1.6-RC3616.jar without the phosphor mod.

**Comment by bluelightning32** [comment]
No repro in spongeforge-1.12.2-2768-7.1.6-RC3625.jar without the phosphor mod.

**Comment by bluelightning32** [comment]
No repro in spongeforge-1.12.2-2768-7.1.6-RC3629.jar without the phosphor mod.

**Comment by bluelightning32** [comment]
The phosphor mod is incompatible with spongeforge-1.12.2-2768-7.1.6-RC3629.jar.

**Comment by bluelightning32** [comment]
Still reproduces in spongeforge-1.12.2-2768-7.1.6-RC3648.jar.

**Comment by gabizou** [comment]
For reference, the boss spawning bits are controlled [here](https://github.com/TeamTwilight/twilightforest/blob/20ec707861c588fb58686e8d229b048acece1143/src/main/java/twilightforest/tileentity/TileEntityTFBossSpawner.java#L37). Will look into it, likely the tile entity is either not getting removed or something else.

**Comment by bluelightning32** [comment]
Confirmed fixed in spongeforge-1.12.2-2768-7.1.6-RC3651.jar.
